app/javascript/components/useIsIntersecting.ts

```diff
@@ -2,9 +2,10 @@ import * as React from "react";
 
 import { useRefToLatest } from "$app/components/useRefToLatest";
 
-export const useIsIntersecting = <T extends HTMLElement = HTMLElement>(
+const useIsIntersecting = <T extends HTMLElement = HTMLElement>(
   callback: (isIntersecting: boolean) => void,
   options?: IntersectionObserverInit,
+  dependencies?: React.DependencyList,
```

Comment:
`useRefToLatest` means we shouldn't need a dependency array (it's also used in a few similar places)

Reply:
Oh I see, interesting! There's probably a clever way to optimize it and avoid observe/disconnect being called so many times, but performance is probably fine in practice so I'm good with it as is